app/src/main/java/org/kiwix/kiwixmobile/localFileTransfer/WifiDirectManager.kt

```diff
@@ -243,8 +243,17 @@ class WifiDirectManager @Inject constructor(
     if (BuildConfig.DEBUG) {
       Log.d(TAG, "Starting handshake")
     }
-    peerGroupHandshakeAsyncTask = PeerGroupHandshakeAsyncTask(this)
-      .also { it.execute() }
+    peerGroupHandshake = PeerGroupHandshake(this)
+      .also {
+        lifecycleCoroutineScope.launch {
+          val inetAddress = it.handshake()
+          inetAddress?.let(::setClientAddress) ?: if (BuildConfig.DEBUG) {
```

Comment:
the entire RHS only executes if we are in debug, does not seem like desired behaviour

Reply:
We discussed that if we have inetAddress null then we want to debug it.

Reply:
and if this happens in a production app we want to do nothing?

Reply:
Ooh! Now I understand. done

Reply:
Ooh! Now I understand. done